EIPS/eip-5344.md

```diff
@@ -0,0 +1,51 @@
+---
+eip: 5344
+title: Soulmates Token
+description: Representing soulmates by collectible soulbound tokens.
+author: Jun Kawasaki (@jun784)
+discussions-to: https://ethereum-magicians.org/t/10132
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-27
+requires: 721, 5114
+---
+
+## Abstract
+
+A soulmates token is a countable soulbound (non-transferrable) non-fungible token that has a fixed list of addresses that collectively own the token.
+
+## Specification
+
+```solidity
+interface IERC5344 is IERC5114 {
+  // returns a current number of mates.
+  // this function **MUST** return the latest number of tokens.
+  function current(uint256 counter) external view returns (uint256)
```

Comment:
Perhaps rename this to `matesOf` or something more descriptive than just `current`? Seems like it would be easy to conflict with other function names as it is currently written.

Reply:
+1, `current` is very overloaded and ambiguous in English